### PR TITLE
Upgrade vte291 to 0.74.2

### PR DESCRIPTION
**Pull request by dmcilvaney**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Update vte291 based on F40, with some modifications to align with AZL. We do not currently offer gi-docgen so I had to disable documentaiton generation for AZL.

###### Change Log  

- Upgrade vte291 to the latest stable release, 0.74.2 (per https://gitlab.gnome.org/GNOME/vte/-/milestones?sort=due_date_desc&state=all 76 seems to be under development still).

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- local
- buddy build: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=533600&view=results

### CODE REVIEW

**Comment by dmcilvaney** [comment]
Waiting on https://github.com/microsoft/azurelinux/pull/8531
Build for combined branches: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=533791&view=results

**Review by jslobodzian — APPROVED** [review]

**Comment by dmcilvaney** [comment]
https://dev.azure.com/mariner-org/mariner/_build/results?buildId=535784&view=results

**Comment by dmcilvaney** [comment]
Build with systemd-bootstrap changes rolled back https://dev.azure.com/mariner-org/mariner/_build/results?buildId=537426&view=results

File: SPECS/vte291/vte291.spec [hunk 1]
```diff
@@ -3,91 +3,139 @@
...
 Requires:       vte-profile
 
+%{!?azl:Conflicts:      gnome-terminal < 3.20.1-2}
+
 %description
 VTE is a library implementing a terminal emulator widget for GTK+. VTE
 is mainly used in gnome-terminal, but can also be used to embed a
 console/terminal in games, editors, IDEs, etc.
 
+%if ! 0%{?azl}
```
Comment (dmcilvaney) [inline SPECS/vte291/vte291.spec:72]:
```suggestion
# We don't have gtk4
%if ! 0%{?azl}
```

**Comment by dmcilvaney** [comment]
Build looks ok

**Pull request merged**